src/components/Trade/hooks/useTradeRoutes/useTradeRoutes.ts

```diff
@@ -19,12 +22,18 @@ export const useTradeRoutes = (): {
   const sellAsset = getValues('sellAsset')
   const assets = useSelector(selectAssets)
   const feeAsset = assets['eip155:1/slip44:60']
+  const asset = useAppSelector(state => selectAssetByCAIP19(state, assetId))
 
   const setDefaultAssets = useCallback(async () => {
     try {
       const [sellAssetId, buyAssetId] = getDefaultPair()
       const sellAsset = assets[sellAssetId]
-      const buyAsset = assets[buyAssetId]
+      // TODO: this is fine for now, since we can only trade ETH/ERC20
+      // but should be refactored to adequate incoming new chains - i.e. thorchain swaps
+      const buyAsset =
+        asset?.chain === 'ethereum' && asset?.caip19 !== sellAssetId
+          ? assets[asset.caip19]
+          : assets[buyAssetId]
```

Comment:
Why do we need chain specific code here? The `caip19` aka `assetId` string is sufficient to identify a specific asset.